### ISSUE TITLE
qterminal -d quits after the last terminal has been closed

### ISSUE THREAD

**Issue by janoff** [qterminal/qterminal#154]
1. open qterminal in dropdown mode
2. in qterminal's bash type "exit" (or press ctrl-d)
=> qterminal process quits

Expected behaviour:
- There should always be at least one active terminal.
- If the last terminal has been exited it is replaced by a new one. (qterminal process can still be stopped by File->Exit)

**Comment by f2404** [comment]
BTW - is this the expected behavior to close an empty terminal tab by pressing Ctrl+D?
As I understand, it's supposed to represent Del; but it also can result in closing the terminal.

**Comment by jleclanche** [comment]
That's normal

**Comment by pmattern** [comment]
*ad Ctrl+D*
AFAIK the expected behaviour of `Ctrl+D` in a terminal emulator is sending EOF. If this happens to be send when no process is running in a shell it happens to end the session.
This is common behaviour and can be seen with Konsole, Xfce4 Terminal and xterm as well.

*ad this issue's actual topic*
Instances of virtually all terminal emulators I'm aware of exit if the last shell session they are providing gets terminated. So I disagree that opening another shell when the last running one is terminated can be called "expected behaviour".
Rather, I'd consider this some pretty uncommon behaviour the implementation of which seems highly questionable.

**Comment by janoff** [comment]
@pmattern This is true as long as we're talking about "qterminal in normal mode". However, this issue (or feature request) is about "qterminal in dropdown mode", so its behaviour should rather be like yakuake's which also doesn't fully shutdown after the last tab has been closed. I suggest using this feature in practice.

**Comment by f2404** [comment]
@janoff Could you check my implementation - is this what you've expected?

**Comment by f2404** [comment]
@pmattern I think I agree with @janoff 
I'm not using drop down terminals myself, but if I would, I'd also expect it to be always there. Please note that there's still a way to close the terminal by using the menu.

**Comment by janoff** [comment]
@f2404: Thanks, much better! ... except that the tab id is not set to 1. :)

**Comment by f2404** [comment]
@janoff Yes, that's because the closed tabs' ids are not reused. Do you think starting again with 1 would be cleaner?

**Comment by janoff** [comment]
Yep. That's also like yakuake handles it.

**Comment by pmattern** [comment]
If I'm not mistaken the only situation where the feature discussed in this issue is of relevance is a single open shell session with no processes running. Thus its only effect is replacing a single shell session with another one of the very same kind instead of ending the terminal emulator as a whole.

I still cannot see any benefit from this and consider this feature not desirable at all, no matter whether or not the behaviour mimics some other tool which btw. goes unmaintained since more than three years.

**Comment by f2404** [comment]
So have we decided that this feature will not be implemented in qterminal?

**Comment by Arnaudv6** [comment]
@pmattern I think starting a new session properly will be better for debugging.
When my terminal bugs, I quit and relaunch.
I expect the same workflow in dropdown mode.
Yakuake -unmaintained be it- got this right I believe.

**Comment by alandmoore** [comment]
Don't know if this feature request has any hope, but FWIW I came here to make this exact request after recently switching to qterminal.  To my thinking, the point of a dropdown terminal is to be an "always on" terminal that's just a keystroke away.  When you hit one too many ctrl-d or type one too many `exit`s, it's a bit disconcerting when that magic key no longer does what you expect.

Thanks!

**Comment by pmattern** [comment]
> To my thinking, the point of a dropdown terminal is to be an "always on" terminal that's just a keystroke away.

Sounds like a reasonable argument to me.
All in all I do think the feature discussed in this issue may very well make sense as far as drop-down mode is concerned by now. So @cja451, @Arnaudv6 and @alandmoore you may want to follow #208 and help testing if needed.

**Comment by ngoettin** [comment]
Recently I had to switch from yakuake to qterminal, because yakuake can't handle screen scaling properly. I like qterminal very much and it would be an almost perfect replacement, if I didn't have to restart it everytime I close the last shell in dropdown mode. For users with the same problem, here is a systemd service that will automatically start qterminal in dropdown mode on login and restart it, when it's closed:

```
~/.config/systemd/user/qterminal.service
----------------------------------------
[Unit]
Description=QTerminal dropdown

[Service]
Type=simple
ExecStart=/usr/bin/qterminal -d
Restart=always

[Install]
WantedBy=default.target
```

This has to be enabled as a user service: `systemctl --user enable qterminal.service`.

I hope this helps someone while this feature request is still open.

**Comment by tsujan** [comment]
IMO, the app should exit with `exit` — as it does now — and being in the drop-down mode shouldn't make a difference.

**Comment by ngoettin** [comment]
I can definitely see your point and I respect your opinion, but it seems to me that the kind of behaviour I mentioned is not too exotic. I think it's the standard behavior of yakuake and it is at least easily configurable in tilda.

The reason is the following: A normal terminal window is expected to be closed when the last shell is closed, but a dropdown terminal is often used as a kind of "always on" terminal that is always only one button press away. It happens often enough to me that I close a terminal and then need a new one some time afterwards. With the current behaviour (if I didn't write the service) I'd need to enter the program name and start it again, which is more work than just pressing F12 again. Having this feature as a toggleable option would be very nice.

After all, not having this feature is not the end of the world and qterminal will still be my new favourite terminal, but having the option would be great.

Thank you for your amazing work!

**Comment by cjaone** [comment]
Even though I think it's pointless to still discuss it after **7 years**, I think a configuration option would be the best thing to do. Anyway, I just switched to Yakuake which does what I want.

**Comment by ngoettin** [comment]
Well the issue is still open :D

Yakuake was great until I got one of these shiny new UHD screens at work and figured out that yakuake can't handle screen scaling

**Comment by tsujan** [comment]
Please note that I started my comment with "IMO". My opinions don't necessarily reflect what may be done in LXQt, especially not here.

There are different habits, and there may be different apps for each.
